dll/win32/kernel32/CMakeLists.txt

```diff
@@ -116,11 +116,40 @@ elseif(ARCH STREQUAL "arm")
         client/arm/thread.S)
 endif()
 
+set(COMPILED_MESSAGE_FILES "")
+
+# Compile message file
+function(compile_message_file MESSAGE_FILE CODEPAGE)
+    get_filename_component(MESSAGE_NAME ${MESSAGE_FILE} NAME_WE) # Filename without .extension
+    set(HEADER_FILE ${CMAKE_CURRENT_BINARY_DIR}/${MESSAGE_NAME}.h)
+    set(RESOURCE_FILE ${CMAKE_CURRENT_BINARY_DIR}/${MESSAGE_NAME}.rc)
+
+    # Custom command with using windmc.exe
+    add_custom_command(
+        OUTPUT ${HEADER_FILE} ${RESOURCE_FILE}
+        COMMAND windmc -h ${CMAKE_CURRENT_BINARY_DIR} -r ${CMAKE_CURRENT_BINARY_DIR} ${MESSAGE_FILE} -C ${CODEPAGE}
+        COMMENT "Compiling ${MESSAGE_FILE} with windmc.exe")
+    add_dependencies(${HEADER_FILE} ${MESSAGE_FILE})
+    add_dependencies(${RESOURCE_FILE} ${MESSAGE_FILE})
+
+    # Add header file and resource file to target
+    set(GENERATED_FILES ${HEADER_FILE} ${RESOURCE_FILE})
+    set_source_files_properties(${GENERATED_FILES} PROPERTIES GENERATED TRUE)
+    list(APPEND COMPILED_MESSAGE_FILES ${GENERATED_FILES})
+    set(COMPILED_MESSAGE_FILES ${COMPILED_MESSAGE_FILES} PARENT_SCOPE)
+endfunction()
+
+compile_message_file(${CMAKE_CURRENT_SOURCE_DIR}/winnls/lang/bg-BG.mc 1251)
+compile_message_file(${CMAKE_CURRENT_SOURCE_DIR}/winnls/lang/de-DE.mc 1252)
+compile_message_file(${CMAKE_CURRENT_SOURCE_DIR}/winnls/lang/pl-PL.mc 1252)
+compile_message_file(${CMAKE_CURRENT_SOURCE_DIR}/winnls/lang/ru-RU.mc 1251)
```

Comment:
What about converting these mc files into utf-8, and then using the existing `add_message_headers` cmake macro as done e.g. here: https://git.reactos.org/?p=reactos.git;a=blob;f=base/system/diskpart/CMakeLists.txt;hb=9a093ecbe93a49ceb86d7599480f6f9f32077823#l51 (line 51) (note that you can specify a list of files to this macro, see this other example: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/mc/CMakeLists.txt;hb=9a093ecbe93a49ceb86d7599480f6f9f32077823#l18 )